### PR TITLE
Jetpack Kotlinization:  JetpackConnectionSource

### PR DESCRIPTION
**Pull request by zwarm**
Parent: #14904 

This PR converts `JetpackConnectionSource` from java to Kotlin
I also added a test, but it might very well be overkill. :)

To test:
- Launch the App
- Login with a site that doesn't have Jetpack installed (I used a pressable site)
- Navigate to stats
- The Jetpack required view should be shown
- Follow the prompts to install Jetpack 
- Follow the prompts for connecting to WP
- Enter valid credentials
- When prompted, allow connection
- App should load as normal

## Regression Notes
1. Potential unintended areas of impact
Jetpack Connections

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual test a Jetpack connection

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/ui/JetpackConnectionSourceTest.kt
Comment (ParaskP7) [inline WordPress/src/test/java/org/wordpress/android/ui/JetpackConnectionSourceTest.kt:7]: 1) Praise (❤️): Kudos for adding this test suite!
2) Suggestion (💡): For next time, consider adding the test before making the refactor, since this will give you much more confidence that the refactor succeeded.
3) Minor (🔍): Consider moving the `Assertions` as an import.
4) Minor (🔍): Consider splitting the `given/when/then` logic with an empty line (the `value/result/assertThat` lines).

File: WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt:3]: 1) Suggestion (💡): I usually suggest to have the construction fields in a separate line instead. I usually prefer to do that for every class. However, for `data` and `enum` classes I am more intrigue in doing so. This is merely me trying to share what I usually do in such case, not insisting on you changing this to my style, just me trying to sync with you.
2) Suggestion (💡): However, for the actual enums themselves (`NOTIFICATIONS` and `STATS`, I would suggest we do keep them in separate lines. Wdyt?

File: WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt]: Suggestion (💡): This `toString()` function can be further simplified to: `override fun toString() = value`

File: WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt]: 1) Suggestion (💡): I suggest having the `@JvmStatic` annotation on a separate line.
2) Suggestion (💡): This `fromString(...)` function can be further simplified to:
```
        fun fromString(value: String) = when {
            NOTIFICATIONS.value == value -> NOTIFICATIONS
            STATS.value == value -> STATS
            else -> null
        }
```

**Review by ParaskP7 — COMMENTED** [review]
👋 @zwarm !

I have reviewed this PR, everything LGTM, good job! 🌟 

However, I didn't manage to test it. I tried to followed the instruction but I am not sure how to create a pressable site without Jetpack. I tried creating new `Pressable` site, it took me a good 1 hours playing with it, but I  failed in the end. Can you help by pairing with me in figuring this out? This knowledge sharing session will be good for me going forward as well. 🙏 

I have left few suggestions (💡) and a couple of minor (🔍) comments for you to consider. I am **NOT** going to approve this PR for now since I haven't tested it, but it seems to me that none of that is blocking so I am just going to `Comment` for now.

File: WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt:3]: > I usually suggest to have the construction fields in a separate line instead. I usually prefer to do that for every class.

I believe the code base leans more towards constructor fields on the same line, especially when it's a single param.

> for the actual enums themselves (NOTIFICATIONS and STATS, I would suggest we do keep them in separate lines.

Again, the code base uses both syntax. 

Perhaps this might be a good discussion for our Android tech chat and see if we want to standardize on some of these things. Neither way is wrong, what reads easier for one person, may read harder for another. I bet we'll get 50 different opinions. :)  At least let's bring it up in our next Raven sync and see if the 3 of us can come to an agreement.

Thanks for sharing :)

File: WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt]: Ugh, running the code reformat moves everything to a separate line again.

File: WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt
Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt:3]: > I believe the code base leans more towards constructor fields on the same line, especially when it's a single param.

> Again, the code base uses both syntax.

> Perhaps this might be a good discussion for our Android tech chat and see if we want to standardize on some of these things. Neither way is wrong, what reads easier for one person, may read harder for another. I bet we'll get 50 different opinions. :) At least let's bring it up in our next Raven sync and see if the 3 of us can come to an agreement.

Yes, you are right, I am going to add it to the backlog, thanks for the idea! 💡 

> Neither way is wrong, what reads easier for one person, may read harder for another.

Actually, in this case, at least version control wise, having those in separate lines is a better alternative and it is no longer that much subjective. If you get those on separate lines and then you choose to update something specific, then the diff will only show that specific update, where is actually makes sense. For example, if you update `NOTIFICATIONS("notifications")` to `NOTIF("notif")`, and have that logic in a separate line, then you wouldn't care about reading `STATS("stats")` in the diff. However, if those are on the same line then you are inevitable forced to review the `STATS("stats")` as well.

The same applies for having the constructor fields in separate line. Actually, the `trailing comma` feature of Kotlin 1.4 is useful because it is basing its usefulness on the same premise, to have as less diff on version control as possible.

> Thanks for sharing :)

Thank YOU! ❤️

File: WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt
Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt]: 🤔 What do you mean Annmarie, we can double check on that by pairing.

Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt]: If you meant that reformatting moves the `@JvmStatic` annotation on the same line, then this is an IDE settings that can be fine tuned.

Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt]: See 57d5ab0

File: WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt]: See 11837b4

Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt]: 👋 @zwarm ! This is probably the wrong commit, right?

File: WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt
Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt]: 👋 @zwarm ! This is probably the wrong commit, right?

Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt]: Also, if you meant that reformatting moves the `when` statement lines, then this is because the `braces` need to be removed as well.

Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt]: > If you meant that reformatting moves the @JvmStatic annotation on the same line, then this is an IDE settings that can be fine tuned.

This is the IDE setting you might need to tune: `Editor` -> `Code Style` -> `Kotlin` -> `Wrapping and Braces` -> `Function annotations` -> `Wrap if long`

**Review by ParaskP7 — APPROVED** [review]
👋 @zwarm !

> However, I didn't manage to test it. I tried to followed the instruction but I am not sure how to create a pressable site without Jetpack. I tried creating new Pressable site, it took me a good 1 hours playing with it, but I failed in the end. Can you help by pairing with me in figuring this out? This knowledge sharing session will be good for me going forward as well. 🙏

**UPDATE**: I actually managed to just test this by deactivating the `Jetpack` plugin on my newly created Pressable site, which activated this plugin automatically. By doing just that I was able to see this screen and verify that this functionality works as expected and described on your instructions.

I am now going to approve this PR, but feel free to update any of the remaining suggestions before merging this to `develop`.

**Pull request merged**
